### PR TITLE
Switch to stack-climbing in dots_capture()

### PR DESCRIPTION
**Pull request by lionel-**
To remove last remnants of promise-climbing in the package.

Includes #11.

### CODE REVIEW

**Comment by lionel-** [comment]
@hadley now rebased.

File: R/dots.R [hunk 1]
```diff
@@ -107,14 +162,16 @@ dots_enumerate_args <- function(dots) {
...
-is_dot_nm <- function(nm) {
-  grepl("\\.\\.[0-9]+$", nm)
+#' Is object a ..n symbol?
+#' @param x An object to test.
+#' @export
+#' @examples
+#' is_dot_symbol(quote(..2))
+#' is_dot_symbol(quote(sym))
+is_dot_symbol <- function(x) {
+  is_symbol(x) && grepl("\\.\\.[0-9]+$", as.character(x))
```
Comment (hadley) [inline R/dots.R]: Doesn't this need something to anchor the match at the start?

Comment (hadley) [inline R/dots.R]: And should use `is_dot_name()`?

Comment (lionel-) [inline R/dots.R]: oops yes

**Pull request merged**